feat(puara): Add PCAAvnd processor

pyandcpp-coder opened the pull request:
This processor performs Principal Component Analysis (PCA), a powerful and widely used technique for dimensionality reduction. It analyzes a dataset to find the principal axes of variance—the directions in which the data is most spread out. The processor outputs the vectors corresponding to these principal components.
By projecting the original data onto the first few principal components, one can reduce the dimensionality of the data while retaining the most important information. This is extremely useful for data visualization, feature engineering for other machine learning tasks, and signal denoising.

pyandcpp-coder commented:
<img width="865" height="748" alt="image" src="https://github.com/user-attachments/assets/b6708bd7-0e31-4735-8a7a-bfd56188602c" />


So here's what I did for the PCA test:

Generated Perfectly Correlated Data
First, I created a simple dataset where the points lie exactly along the line y = 2x.

For the X values, I just used a simple loop: i * 0.1, so I got [0, 0.1, 0.2, ...].

For Y, I used i * 0.2, which gives me values that are exactly double the Xs — maintaining that y = 2x relationship.

Then, I used the Intersperse mode in the Array Combiner to mix them into a single stream: [x1, y1, x2, y2, ...]. That’s the format needed for the PCA processor.

Ran PCA on the Data
I configured the PCA processor with 2 input features (since it’s 2D data) and asked it to find just 1 principal component, the most important direction.

Got the Output & Verified It
The output I got was: [-0.45, -0.89].
And that’s spot on — that’s a unit vector pointing in the direction of the line y = 2x.
Mathematically, the correct direction is [-0.447, -0.894], so my result is basically perfect, just rounded a bit. The negative sign is fine too — PCA doesn’t care about direction, just the axis of variance.

pyandcpp-coder commented:
@jcelerier , Processor and Docs given. Kindly let me know if any changes

pyandcpp-coder commented on CMakeLists.txt:
```diff
@@ -75,30 +71,28 @@ avnd_score_plugin_add(
 avnd_score_plugin_add(
   BASE_TARGET score_addon_puara
   SOURCES
-    Puara/WalkerAvnd.hpp
```
Ya, Fixed now.

jcelerier commented on Puara/PCAAvnd.cpp:
```diff
@@ -0,0 +1,47 @@
+  if (n_samples < n_features) {
+    return;
+  }
+  using MatrixType = Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
```
since we are doing some real-time computations, this is not very efficient as it will allocate memory on every run. Instead we'd really want to store them as members of the class and reuse memory as far as possible

pyandcpp-coder replied:
Yes, and actually I was working on that trying to learn how i could optimize this processor

pyandcpp-coder commented:
@jcelerier ,
Actually I was working on optimization for this processor
Here's what I optimized for real-time performance:

1. Moved all Eigen matrices (MatrixXd, SelfAdjointEigenSolver) from local variables to class members
2. Added dimension caching (m_cached_n_samples, m_cached_n_features) to track when resizing is needed
3. Only resize matrices when input dimensions actually change, not on every call
